Update templating docs (#272)
MicrosoftDocs/AdaptiveCards: AdaptiveCards/templating/language.md

> [!IMPORTANT] 
> 
> **Breaking changes** in the **May 2020 Release Candidate**
>
> We've been hard at work getting templating released, and we're finally in the home stretch! We had to make some minor breaking changes as we close on the release.

> ## Breaking changes as of May 2020

Reviewer: Ideally, these would reference the Release notes which would have the breaking changes listed out to avoid redundancy. Any objection to making that quick change here once we do have the preview github release note published next week?

1. The binding syntax changed from `{...}` to `${...}`
    * For Example: `"text": "Hello {name}"` becomes `"text": "Hello ${name}"`
    
## Binding to data

When authoring a template you have two options to provide to relevant data.

1. **Option A: Inline with the template payload**. You can provide the data inline with the `AdaptiveCard` template payload. To do so, simply add a `$data` attribute to the root `AdaptiveCard` object.
2. **Option B: As a separate data object**. with this option you provide two separate objects to the [Templating SDK](sdk.md) at runtime: the template and the data. This will be the more common approach, since typically you will create a template and want to provide dynamic data later.

> ### Optioan A: Inline data

Reviewer 2: typo: Option

You can bind to the data within the `body` or `actions` of the card.

> 

Reviewer 2: I feel like this paragraph makes a jump out of nowhere, probably adding a subtitle similar to "Features supported" or some kind of introduction may make more sense

    "body": [
        {
            "type": "TextBlock",
            "text": "Hi ${employee.name}! Here's a bit about your org..."
        },
        {
            "type": "TextBlock",
            "text": "Your manager is: ${employee.manager.name}"
        },
        {
            "type": "TextBlock",
            "text": "3 of your peers are: ${employee.peers[0].name}, ${employee.peers[1].name}, ${employee.peers[2].name}"
        }
    ]
}
```

### Option B: Separating the template from the data

> Alternatively (and more likely), you will create a re-usable card "template" without including the data. This template could be stored as a file and added to source control.

Reviewer: Reformed using [contractions](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=28361) ("you'll" instead of "you will")
```suggestion
Alternatively (and more likely), you'll create a re-usable card template without including the data. This template could be stored as a file and added to source control.
```

> This part is a bit of "dark magic". Feedback welcome.

Reviewer 2: is the feedback section staying? It feels kind of weird to read this while going through the docs

Reviewer: I agree - also we should remove the usage of "dark magic" from our public documentation :)

}
```

## Built-in functions

> No templating language is complete without a rich suite of helper functions. Adaptive Card Templating is built on top of the [Adaptive Expression Language](https://github.com/microsoft/BotBuilder-Samples/tree/master/experimental/common-expression-language) (AEL), which is an open standard for declaraing expressions that can be evaluated on many different platforms. And it's a proper superset of "Logic Apps", so you can use similar syntax as Power Automate, etc.

Reviewer: ```suggestion
No templating language is complete without a rich suite of helper functions. Adaptive Card Templating is built on top of the [Adaptive Expression Language](https://github.com/microsoft/BotBuilder-Samples/tree/master/experimental/common-expression-language) (AEL), which is an open standard for declaring expressions that can be evaluated on many different platforms. And it's a proper superset of [Microsoft Logic Apps](https://azure.microsoft.com/en-us/services/logic-apps/), so you can use similar syntax as [Microsoft Power Automate](https://flow.microsoft.com/en-us/), etc.
```
